envoy/http/filter.h

```diff
@@ -857,7 +857,7 @@ class StreamFilterBase {
    * onDestroy() invoked. Filters that cross-register as access log handlers receive log() before
    * onDestroy().
    */
-  virtual void onDestroy() PURE;
+  virtual void onDestroy() {}
```

Comment:
Can you keep this method pure virtual if you add override onDestroy in the PassThroughFilter? Or is it causing more build errors than that?

Reply:
Thanks Yan, but I think if we do that, an existing class will not compile if
- It derives directly from `PassThroughDecoderFilter` or `PassThroughEncoderFilter`, and
- It doesn't override onDestroy, i.e. it relies on the empty implementation in `PassThroughDecoderFilter` or `PassThroughEncoderFilter`.

Reply:
/assign @yanavlasov

Reply:
Did you try it? Can you post the compiler error, please?

Reply:
As discussed, I change to have 3 `onDestroy` implementations in `PassThroughDecoderFilter`, `PassThroughEncoderFilter`, and `PassThroughFilter`. PTAL.